### PR TITLE
Fix docs, add links in README.md, THEORY.md

### LINKED ISSUE #8: Changes in Theory.md
Hey there!

I'd like to make the following changes to the THEORY.md file :
- Markdown in the ordered list and superscript
- Add links to the papers mentioned
- Add a summary of each of the three theories

And in the README.md file, I'd like to:
- Reflect the changes in markdown
- Add a link to the THEORY.md file for more details

I think that this would add value to the project by way of documentation and thus, would like to create a PR on this issue...
I hope that's fine!

### PR DESCRIPTION
**Pull request by Akshu-on-github**
In reference to issue #8 
Unfortunately, I wasn't able to summarize the mentioned theories to my satisfaction, will try to do so at a later date

### CODE REVIEW

**Comment by nbcl** [comment]
**🎉 Thanks for your interest in contributing to Primality**

Hello @AkshayaKulasekaran! 

I just reviewed the request, it looks good!

There seem to be a minor issue in the provided links, as they do not correspond to the correct source.

For example, the third quote references the OEIS Foundation but the url redirects to Wikipedia.

I suggest the provided link to each of the citations comes from each of the following websites respectively:

1. For the first source: https://dl.acm.org/
2. For the second source: https://www.ams.org/
3. For the third source: https://oeis.org/
 
If you have trouble finding the correct sources, you can let me know by removing the hyperlinks and adding this change to your fork. If that is the case, I will move foward with the request and the correct sources shall be added on a later date.

**Thank you!**

**Review by nbcl — COMMENTED** [review]
Change citation hyperlinks to represent correct source

File: theory/THEORY.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ## Theory
 
-The current library uses the Miller primality test (1) with known strong pseudoprime bases (2) to provide deterministic testing for integers between 2 and 2^64 - 1. (3)
+The current library uses the Miller primality test<sup>(1)</sup> with known strong pseudoprime bases<sup>(2)</sup> to provide deterministic testing for integers between 2 and 2^64 - 1<sup>(3)</sup>.
 
-- (1) Miller, G. (1976), "Riemann's Hypothesis and Tests for Primality", Journal of Computer and System Sciences.
+1. Miller, G. (1976), ["Riemann's Hypothesis and Tests for Primality"](https://core.ac.uk/download/pdf/82649441.pdf), Journal of Computer and System Sciences.
 
-- (2) Zhang, Z and Tang, M. (2003), "Finding strong pseudoprimes to several bases. II", Mathematics of Computation. 
+1. Zhang, Z and Tang, M. (2003), ["Finding strong pseudoprimes to several bases. II"](http://www.jointmathematicsmeetings.org/mcom/2003-72-244/S0025-5718-03-01545-X/S0025-5718-03-01545-X.pdf), Mathematics of Computation.
```
Comment (thisisjaymehta) [inline theory/THEORY.md:7]: Should this bullet say 2

**Review by thisisjaymehta — CHANGES_REQUESTED** [review]
Bullets should say 1,2 and 3 instead of 1,1 and 1 in Theory Subsection

**Comment by Akshu-on-github** [comment]
Heya @nbcl!
So sorry for the delay, I've redone the links in THEORY.md

Also, @thisisjaymehta, that's actually the mark-up for ordered lists, so yeah... 😅

**Comment by Akshu-on-github** [comment]
> Bullets should say 1,2 and 3 instead of 1,1 and 1 in Theory Subsection

Check out the rich text, it should be a bit clearer

**Comment by Akshu-on-github** [comment]
Ah, one more thing... the modified links redirect to the general website, while the previous ones redirect to the specific papers/ info about those papers. Just pointing that out, in case that wasn't what you wanted.

**Review by thisisjaymehta — APPROVED** [review]

**Comment by nbcl** [comment]
Hello @AkshayaKulasekaran! 

I think there was a small misunderstandment, my apologies.

When I suggested, for example, `https://www.ams.org/` as a source, I didn't mean that exact site, I meant the source document hosted on that specific site. 

A correct citation should look similar to`https://www.ams.org/documents/source.pdf`, where `source.pdf` represents the document you are trying to cite.

Again, if you encounter problems finding the specific sources, I would suggest to remove the hyperlinks so we can move forward with the request! This being the case, the exact sources shall be added on a later date.

**Thank you!**

**Review by nbcl — COMMENTED** [review]
Change or remove citation hyperlinks to represent exact source

**Comment by Akshu-on-github** [comment]
@nbcl, I was able to find the links, I've updated them accordingly

**Review by nbcl — COMMENTED** [review]
**Fixed:** Change citation hyperlinks to represent correct source.

**Review by nbcl — APPROVED** [review]
Approved! ✔️

**Pull request merged**